javarosa/src/main/java/org/javarosa/xpath/expr/XPathFuncExpr.java

```diff
@@ -1449,7 +1449,7 @@ public static Object InferType(String attrValue) {
         }
 
         try {
-            Double ret = new Double(Double.parseDouble(attrValue));
+            Double ret = MathUtils.parseDoubleSafe(attrValue);

```

Comment:
This is at the Javarosa level so would be hard to make contingent on Android version. According to SO and my understanding of regex's this should actually be more performant. Maybe @ctsims can weigh in - can also just manually check for the Infinity value if we think the parseDouble function is more performant.